source/common/stats/thread_local_store.cc

```diff
@@ -72,11 +72,7 @@ void ThreadLocalStoreImpl::removeRejectedStats(StatMapClass& map, StatListClass&
 }
 
 bool ThreadLocalStoreImpl::rejects(StatName stat_name) const {
-  // Don't both elaborating the StatName there are no pattern-based
-  // exclusions;/inclusions.
-  if (stats_matcher_->acceptsAll()) {
-    return false;
-  }
+  ASSERT(!stats_matcher_->acceptsAll());
```

Comment:
Sorry to divebomb here; it might be worth adding a comment noting that we're guaranteed `acceptsAll() == false` by the early return in `checkAndRememberRejection()`.